subtrack.MAUI/Services/SubscriptionService.cs

```
@@ -52,9 +56,24 @@ public async Task Delete(int id)
         public async Task<Subscription> CreateSubscriptionAsync(Subscription subscriptionToCreate)
         {
             subscriptionToCreate.LastPayment = subscriptionToCreate.LastPayment.Date;
+            subscriptionToCreate.FirstPaymentDay = subscriptionToCreate.LastPayment.Day;
             await _context.Subscriptions.AddAsync(subscriptionToCreate);
             await _context.SaveChangesAsync();
             return subscriptionToCreate;
         }
+        public async Task<DateTime> UpdateLastPaymentDateAsync(int subscriptionId, DateTime newLastPaymentDate)
+        {   
+            var sub = await _context.Subscriptions.FirstOrDefaultAsync(s => s.Id == subscriptionId);
+
+            if (sub is not null)
```

if sub is null we want to throw an exception, and by handling that case first our update code does not need to be in the if statement